web/src/shared/ServiceContext.js

```diff
@@ -0,0 +1,96 @@
+/*
+    Copyright (c) QC Coders. All rights reserved. Licensed under the GPLv3 license. See LICENSE file
+    in the project root for full license information.
+*/
+
+import React, { Component } from 'react';
+import Snackbar from '@material-ui/core/Snackbar';
+import api from '../api';
+
+const ServiceContext = React.createContext();
+
+const initialState = {
+	snackbar: {
+        message: '',
+        open: false,
+    }
+}
+
+class ContextProvider extends Component {
```

Comment:
rename file to match, i like `ContextProvider`

Reply:
Makes sense